superset-frontend/packages/superset-ui-core/src/query/types/Query.ts

```diff
@@ -165,4 +165,154 @@ export interface QueryContext {
   form_data?: QueryFormData;
 }
 
+export const ErrorTypeEnum = {
+  // Frontend errors
+  FRONTEND_CSRF_ERROR: 'FRONTEND_CSRF_ERROR',
```

Comment:
where did these come from?

Reply:
From [superset-frontend/src/components/ErrorMessage/types.ts](https://github.com/apache/superset/blob/master/superset-frontend/src/components/ErrorMessage/types.ts#L21), I needed to bring it over to superset-ui for the `SupersetError` type, which was needed in the `Query` type. I did not like copying this much code, but it seemed like the best way to do this for now 😅

Reply:
oh I see.. what happened when you tried importing the Query type from src/SqlLab/types?

Reply:
It gives a few errors:

'superset' should be listed in the project's dependencies. Run 'npm i -S superset' to add it

File '/Users/lyndsikaywilliams/Preset/incubator-superset/superset-frontend/src/SqlLab/types.ts' is not under 'rootDir' '/Users/lyndsikaywilliams/Preset/incubator-superset/superset-frontend/packages/superset-ui-core/src'. 'rootDir' is expected to contain all source files.

File '/Users/lyndsikaywilliams/Preset/incubator-superset/superset-frontend/src/SqlLab/types.ts' is not listed within the file list of project '/Users/lyndsikaywilliams/Preset/incubator-superset/superset-frontend/packages/superset-ui-core/tsconfig.json'. Projects must list all files or use an 'include' pattern.